test/karma/providers/viewBindingsSpec.js

```diff
@@ -0,0 +1,173 @@
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', {
+          'bindings' : [{ requiredState: ['a']}, { requiredState: ['b']}],
+          'commonRequiredState': 'c'
+        }
+        );
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([{ requiredState: ['a', 'c']}, { requiredState: ['b', 'c']}]));
+  });
+
+    it('when commonRequiredState is a string and existing requiredState is a string', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', {
+          'bindings' : [{ requiredState: 'a'}, { requiredState: 'b'}],
+          'commonRequiredState': 'c'
+        }
+        );
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([{ requiredState: ['a', 'c']}, { requiredState: ['b', 'c']}]));
+  });
+
+    return it('when commonRequiredState is an array and existing requiredState is an array', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', {
+          'bindings' : [{ requiredState: ['a']}, { requiredState: ['b']}],
+          'commonRequiredState': ['c', 'd']
+        });
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([{ requiredState: ['a', 'c', 'd']}, { requiredState: ['b', 'c', 'd']}]));
+  });
+});
+
+  describe('commonResolve:', function() {
+
+    it('merges with existing resolve section', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', {
+          'bindings' : [ { resolve: { 'a' : 'a', 'b': 'b' } } ],
+          'commonResolve': { 'b' : 'override', 'c': 'c' }
+        });
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([{ resolve: { 'a': 'a', 'b': 'b', 'c': 'c' } }]));
+  });
+
+    return it('creates resolve section if missing', function() {
+
+      window.angular.mock.module(function(ViewBindingsProvider) {
+        ViewBindingsProvider.bind('name1', {
+          'bindings' : [ {} ],
+          'commonResolve': { 'a' : 'a' }
+        });
+      });
+
+      return inject(ViewBindings => expect(ViewBindings.getView('name1').getBindings()).toEqual([{ resolve: { 'a': 'a' } }]));
+  });
+});
+
+  describe('Application of basic common fields', function() {
+    let basicCommonFields = [
+      { name: 'commonResolvingErrorTemplateUrl', overrideField: 'resolvingErrorTemplateUrl' },
+      { name: 'commonResolvingErrorComponent', overrideField: 'resolvingErrorComponent' },
+      { name: 'commonErrorComponent', overrideField: 'errorComponent' },
+      { name: 'commonErrorTemplateUrl', overrideField: 'errorTemplateUrl' }
+    ];
+
```

Comment:
Array.from(xxx).map is quite odd here... please just loop over the fields, preferably using lodash